scenes.go

```diff
@@ -316,14 +316,30 @@ func scene6() (*sceneConfig, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	newGeometries := make([]*referenceframe.GeometriesInFrame, 0)
+	oldWorld, err := cfg.WorldState.ToProtobuf()
+	for _, protoGeometries := range oldWorld.GetObstacles() {
+		oldGeometries, err := referenceframe.ProtobufToGeometriesInFrame(protoGeometries)
+		if err != nil {
+			return nil, err
+		}
+		newGeometries = append(newGeometries, oldGeometries)
+	}
+
 	obstacle, err := spatialmath.NewBox(spatialmath.NewPoseFromPoint(r3.Vector{-150, 0, 0}), r3.Vector{20, 2000, 2000}, "extra_obs")
 	if err != nil {
 		return nil, err
 	}
-	cfg.WorldState.Obstacles = append(
-		cfg.WorldState.Obstacles,
-		referenceframe.NewGeometriesInFrame(referenceframe.World, []spatialmath.Geometry{obstacle}),
-	)
+	obstaclesInFrame := referenceframe.NewGeometriesInFrame(referenceframe.World, []spatialmath.Geometry{obstacle})
+	newGeometries = append(newGeometries, obstaclesInFrame)
+
+	newWorldState, err := referenceframe.NewWorldState(newGeometries, nil)
```

Comment:
Didn't do much code optimization on this pass, this could probably be trimmed a bit